cmd/entc/entc.go

```diff
@@ -194,15 +194,27 @@ func createDir(target string) error {
 	if err := os.MkdirAll(target, os.ModePerm); err != nil {
 		return fmt.Errorf("creating schema directory: %w", err)
 	}
-	if target != defaultSchema {
-		return nil
-	}
-	if err := ioutil.WriteFile("ent/generate.go", []byte(genFile), 0644); err != nil {
+	if err := ioutil.WriteFile(filepath.Join(target, "/../generate.go"), getGenerateFileData(target), 0644); err != nil {
 		return fmt.Errorf("creating generate.go file: %w", err)
 	}
 	return nil
 }
 
+// ent/generate.go file used for "go generate" command.
+func getGenerateFileData(target string) []byte {
+	pathArr := strings.Split(target, "/")
+
+	var sb strings.Builder
+	sb.Grow(32)
+
+	sb.WriteString("package ent\n\n")
```

Comment:
The generated package name is derived from the target directory (i.e. it's not `package ent`) - see the code in `entc/entc.go`. Maybe we need to export this logic.

Reply:
See if 73ebac3 works as expected.